sdk/monitor/monitor-opentelemetry-exporter/src/export/base.ts

```diff
@@ -123,19 +154,40 @@ export abstract class AzureMonitorBaseExporter {
             });
           }
           if (filteredEnvelopes.length > 0) {
+            if (!this._isStatsbeatExporter && this._statsbeatMetrics) {
+              this._statsbeatMetrics.countRetry(statusCode);
+            }
             // calls resultCallback(ExportResult) based on result of persister.push
             return await this._persist(filteredEnvelopes);
           }
           // Failed -- not retriable
+          if (this._isStatsbeatExporter) {
+            this._statsbeatFailureCount++;
+          } else {
+            if (this._statsbeatMetrics) {
+              this._statsbeatMetrics.countFailure(endTime - startTime, statusCode);
+            }
+          }
           return {
             code: ExportResultCode.FAILED,
           };
         } else {
           // calls resultCallback(ExportResult) based on result of persister.push
+          if (!this._isStatsbeatExporter && this._statsbeatMetrics) {
+            this._statsbeatMetrics.countRetry(statusCode);
+          }
           return await this._persist(envelopes);
         }
       } else {
         // Failed -- not retriable
+        if (!this._isStatsbeatExporter && this._statsbeatMetrics) {
+          if (statusCode) {
+            this._statsbeatMetrics.countFailure(endTime - startTime, statusCode);
+          }
+        }
+        if (this._isStatsbeatExporter) {
+          this._statsbeatFailureCount++;
```

Comment:
Can you shutdown Statsbeat here?, trying to shut it down every time exporter is called will cause some unnecessary overhead, a single check during export will be preferred if possible

Reply:
I can see Failure count could be incremented in different places, add a method to decrease the counter and turn Statsbeat off when limit is reached

Reply:
Makes sense if this is getting called every 15 minutes for every exporter the overhead could get large. Removed the check at the beginning of the export process and moved that to a method that increments failures and then determines if we shutdown statsbeat metric collection. This way the check only gets triggered in the event of a failure.